src/core/State.cpp

```diff
@@ -47,25 +47,29 @@
 #include <QtXmlPatterns/QXmlQuery>
 
 State::State()
-    : version_( INVALID_FILE_VERSION )
+    : displayGroup_( new DisplayGroup( QSizeF( 0, 0 )))
+    , version_( INVALID_FILE_VERSION )
+{}
+
+State::State( DisplayGroupPtr displayGroup )
+    : displayGroup_( displayGroup )
+    , version_( INVALID_FILE_VERSION )
 {
 }
 
-State::State(const ContentWindowPtrs& contentWindows)
-    : contentWindows_( contentWindows )
-    , version_( INVALID_FILE_VERSION )
+ContentWindowPtrs& State::getContentWindows()
 {
+    return displayGroup_->getContentWindows();
 }
 
-const ContentWindowPtrs& State::getContentWindows() const
+DisplayGroup& State::getDisplayGroup()
 {
-    return contentWindows_;
+    return *displayGroup_;
 }
 
 bool State::legacyLoadXML( const QString& filename )
 {
-    contentWindows_.clear();
-
+    displayGroup_->getContentWindows().clear();
```

Comment:
why not fill a local list here and use displayGroup_->setContentWindows()? Exposing a write reference to a class member needs to be questioned.